docs/development/convention.md

```diff
@@ -1,48 +1,108 @@
 # Convention
 
-## C++ Style Guide
+## Styling
+
+
+We follow the standards of the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html), but have adapted some of the conventions to better fit our needs. Differences and more detailed explanations of some rules are highlighted below.
 
 ### Casting
 
-It's allowed to avoid explicit casting when an integral value is used as a constant and
-can be implicitly cast. See [casting](https://google.github.io/styleguide/cppguide.html#Casting).
+If an implicit cast for a constant value integral is valid, do not use a redundant explicit cast.
 
 ```c++
-// Using an integral value as a constant without explicit casting is allowed
 template <size_t N>
 class BigInt {};
 
-BigInt<1> bigint;         // This is okay
-BigInt<size_t{1}> bigint; // This is not allowed
+// DO
+BigInt<1> bigint;
 
-// When using a constant integral value, casting is not necessary
-size_t GetSize() { return 1; } // This is okay
+// DON'T DO
+// This will be casted redundantly as BigInt<size_t{size_t{1}}>
+BigInt<size_t{1}> bigint;
 
-// However, when performing operations like bit-shifting, it's recommended
-// to follow the google c++ style guide.
-size_t size = size_t{1} << bit; // Recommended to cast as per the style guide
+// DO
+size_t GetSize() { return 1; }
+
+// Note: Continue to use explicit casts for cases that require it logically
+size_t size = size_t{1} << bit;
 ```
 
-## GPU
+[Correlating Google C++ casting rule here](https://google.github.io/styleguide/cppguide.html#Casting)
 
-### Naming
+### Comments
 
-- If your gpu codes can be run independent to GPU vendors, please use a word `Gpu` not `Cuda` or `ROCm`.
+#### Naming
 
-### File Suffixes
+Enclose code-specific names in the comments with vertical bars "|". This includes names of variables, functions, files, objects, and more.
 
-- If a file contains kernels, there will be suffixes `.cu.xx` despite it can be run on `ROCm`.
-- If a file contains device only codes, there will be suffixes `.cu.xx` despite it can be run on `ROCm`.
-- If a file contains host and device codes, then suffixes are not appended.
-  For examples, see files under `tachyon/devices/gpu`.
+```c++
+// DON'T DO:
+// "foo" is 10
+// `foo` is 10
+
+// DO:
+// |foo| is 10
+size_t foo = 10;
+```
+
+[Correlating Google C++ commenting rule here](https://google.github.io/styleguide/cppguide.html#Function_Comments)
+
+#### Greek Letters
+
+Greek letters are often used in explanations of mathematical processes in our comments. We use the "Small" version of Greek unicode letters to standardize  and assist searching capabilites.
+
+For example:
+- Use θ (U+03B8 -- "Greek Small Letter Theta")
+- Do not use Θ (U+0398 -- "Greek Capital Letter Theta")
+
+
+### Build Targets
 
-## Commit
+Since styling Bazel files is not specified in the Google C++ style guide, here are the rules we abide by:
+1. Order all `tachyon_cc_unittest` targets after all `tachyon_cc_library` targets.
```

Comment:
Please refer to https://github.com/DavidAnson/markdownlint/blob/v0.33.0/doc/md032.md

```suggestion
Since styling Bazel files is not specified in the Google C++ style guide, here are the rules we abide by:

1. Order all `tachyon_cc_unittest` targets after all `tachyon_cc_library` targets.
```